### PR TITLE
Make `./mach test` work with unit tests

### LINKED ISSUE #7618: Make `mach test` work with Rust unit tests
We should enable `./mach test tests/unit/net/http_loader.rs` to make it really easy for people to execute unit tests. This would be converted internally to `./mach test-unit -p net`; if there's some way to only execute the tests for the `http_loader` module, that would be even better.

Code: python/servo/testing_commands.rs

### PR DESCRIPTION
**Pull request by 6112**
Fixes #7618.

Allows running a specific Rust test with one of:

```
./mach test tests/unit/net/http_loader.rs

./mach test-unit tests/unit/net/http_loader.rs
./mach test-unit tests/unit/net/http_loader
./mach test-unit net/http_loader
```

Allows running a whole package's tests with one of:

```
./mach test tests/unit/net

./mach test-unit tests/unit/net
```

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7809)

### CODE REVIEW

**Comment by jdm** [comment]
This looks like a great improvement, @6112! Thanks!

r? @frewsxcv

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -146,13 +149,35 @@ def test_unit(self, test_name=None, package=None):
...
-            packages = [package]
+            packages = {package}
         else:
-            packages = os.listdir(path.join(self.context.topdir, "tests", "unit"))
+            packages = set()
+
+        test_patterns = []
+        for test in test_name:
+            # add package if 'tests/unit/<package>'
+            match = re.search("tests/unit/(\\w+)/?$", test)
```
Comment (frewsxcv) [inline python/servo/testing_commands.py]: FYI, there's an [`fnmatch`](https://docs.python.org/2/library/fnmatch.html) library built into Python that would probably be more appropriate for matching filenames, but this is simple enough that it doesn't really matter

Comment (frewsxcv) [inline python/servo/testing_commands.py]: Nevermind. I don't think there's anything in `fnmatch` that allows for capturing of text within filepaths like you're doing here

**Comment by frewsxcv** [comment]
Looks good, thanks!

@bors-servo r+

**Pull request merged**